core-rust/state-manager/src/state_manager.rs

```diff
@@ -282,7 +279,7 @@ impl<S: ReadableStore> StateManager<S> {
                     store,
                     &self.scrypto_interpreter,
                     &self.fee_reserve_config,
-                    &self.execution_config,
+                    if parent_transaction_identifiers.state_version == 0 { &self.execution_config_for_genesis } else { &self.execution_config },
```

Comment:
(nitpick, just to remember during next refactor here)
 ```suggestion
                    self.get_execution_config(parent_transaction_identifiers),
```

Reply:
Are you suggesting that we add this method `get_execution_config`?

This is sadly a classic rust-ism.

What you propose doesn't actually work - because of how Rust handles reference aliasing - it wouldn't allow (eg) a mutable reference to `self.execution_cache` at the same time as calling a method `self.get_execution_config(..)` on a shared reference &self, because it doesn't know that `self.get_execution_config` doesn't read from `self.execution_cache`, violating the mutable reference uniqueness.

Or am I misunderstanding you? :)

Reply:
> Or am I misunderstanding you? :)

no, I apparently meant to commit a rust-ism :p I understand now, let's live with the random `if` 👍 
(I do have a https://github.com/radixdlt/babylon-node/pull/364 pending, which touches the same lines and awaits a nasty merge; I'll see if I can do anything about it there)